### DISCUSSION TITLE
Video capture Live Video webpage

### DISCUSSION THREAD

**Discussion by d4rkdpg** [awawa-dev/HyperHDR#475] [General]
Hi thanks for HyperHDR and your efforts.  I've spent most of the holidays setting it up - works perfectly.

Is it possible to provide the link (url) for the live video stream - I want to feed it into my home assistant instance as a webpage?

A bit of details on my setup:

- WS2812B LED Strips for TV and TV cabinets (left and right of TV)
- ESP8266 with Hyper SPI used connected to raspberry pi 4 
- TV - primary instance
- TV cabinets  - secondary instance using RpiPWM (ws8281x) so I can switch between tv capture and static colours/effects
- ms2109 hdmi usb capture card connected to raspberry pi
-  HDMI 2.1 Switch 8K, AVIDGRAM HDMI Switcher 2  - PS5 and google chromecast (with autoswitching enabled)
-  Escoo 4K 120Hz HDMI 2.1 Splitter 1 in 2 Out 8K  - 1 out 4k 120hz to TV 2nd Out (switched to 1080p) to hdmi usb capture

I have a few issues with 120hz (PS5) sometimes I get signal loss (rainbow colours) - in logs I've noticed that I have to enable sparse processing, anything else I can do about this?

Thanks

**Comment by awawa-dev** [comment]
> Is it possible to provide the link (url) for the live video stream - I want to feed it into my home assistant instance as a webpage?

Not sure about it...do you want to feed HyperHDR or get HyperHDR's stream? First is possible using Flatbuffer/Protobuffer communication. Second requires integration with the API (JPEG stream).

> I have a few issues with 120hz (PS5) sometimes I get signal loss (rainbow colours) - in logs I've noticed that I have to enable sparse processing, anything else I can do about this?

It's a hardware, not software problem. Sparse processing is not related and it automatically kicks in when non-optimal user configuration is detected.

Moving to the discussion.

**Reply by d4rkdpg** [reply]
thanks for replying, I want to grab the stream and display it somewhere else, not hyperhdr as a stream in a webpage.

**Reply by awawa-dev** [reply]
There is no easy way to do it, but you can analyze the code of the video preview: https://github.com/awawa-dev/HyperHDR/blob/master/assets/webconfig/js/ledsim.js

**Comment by d4rkdpg** [comment]
Will check it out.  Reference hardware are you referring to the capture card?  Would real  a usb 3  work better in this case.

**Reply by awawa-dev** [reply]
The test available on the blog shows lower latency when using USB 3.0 I'm currently using the Elgato HD60x which guarantees great quality and I don't have any problems with HDMI handshake or anything else (but I don't use HDCP or CEC which may be important for some users): you get what you pay for. But I also had no problems with MS2109/Feintech, although I don't use consoles or 120Hz mode either.

**Comment by jonoscott** [comment]
I'm very curious about this too, I've been run in Hyperion on and RPi3 for a couple of years now and have had the live video stream coming from that from the Integration for HA and had it setup on my Dashboard for remotely controlling the TV when I'm not actually in front of it... Was setup as a Camera feed.

I've now replaced my Pi with something a little more grunty and installed HyperHDR on it which works great, it's just missing that one feature which I love.

**Comment by TechFreak93** [comment]
I'm also looking for a way to integrate the LED preview in the HomeAssistant via URL link.
It would work much better than the integration of Hyperion.
I also did it this way, only after each reload of the HA page I have to manually switch back to the LED preview.
I don't always want to have to turn on the projector when listening to music and with small settings.
And HyperHDR makes the best ambience in my home cinema with over 700 LEDs.
It would be great and I would be very grateful if there was a possibility for that. 
 
![Screenshot_20240506_185813_Home Assistant](https://github.com/awawa-dev/HyperHDR/assets/166557737/f563955a-87af-427a-aa83-c3ecd97711a3)

**Comment by leoravagnani** [comment]
Just found about HyperHDR and Hyperion - which is lacking. HyperHDR seems so much better! 

Hyperion's official Home Assistant integration provides a stream of the source video as a camera feed.
Would be nice if HyperHDR was able to do the same, so we can configure and troubleshoot our TVs remotely (not that we should - but you, it's the home automation hell hole).

I'm using this custom integration https://github.com/mjoshd/hyperhdr-ha

**Reply by awawa-dev** [reply]
Hi
Unfortunately, I'm not sure what the status of the @mjoshd plugin project is, whether it's still in active development. I am not involved in the maintenance of this project.
